packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -131,9 +148,34 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    * handles disliking the generated answer.
    * @param {CustomEvent} event
    */
-  handleDislike(event) {
+  async handleDislike(event) {
     event.stopPropagation();
     this.generatedAnswer.dislike?.();
+    if (!this.feedbackSubmitted) {
+      // @ts-ignore
+      await FeedbackModal.open({
+        label: this.labels.feedback,
+        size: 'small',
+        description: this.labels.feedback,
+        options: this.options,
+        handleSubmit: this.submitFeedback.bind(this),
+        optionsLabel: this.labels.whyGeneratedAnswerWasNotHelpful,
+      });
+      this.generatedAnswer.closeFeedbackModal();
+    }
+  }
+
+  /**
+   * Submits the feedback
+   * @returns {void}
+   */
+  submitFeedback(feedbackPayload) {
```

Comment:
Question: what happens if a user clicks on submit in the feedback modal but hasn't selected a value or entered details? Does it still submits with empty feedback?

Reply:
No the feedback is not submitted, the inputs are required so we can't submit without providing a value and we display an error message like the following to inform the user about that:
<img width="1711" alt="invalid" src="https://github.com/coveo/ui-kit/assets/86681870/5fbafe5a-d83a-4aea-8d16-cadd49f80490">

<img width="1544" alt="invalid2" src="https://github.com/coveo/ui-kit/assets/86681870/ef8c2947-acac-4931-a2ea-bb968107f049">